### ISSUE TITLE
Problems with type Integer, values > 2^31 - 1 

### ISSUE THREAD

**Issue by jschoenbohm** [OpenModelica/OpenModelica#7354]
This issue belongs to the ticket: [(<https://trac.openmodelica.org/OpenModelica/ticket/6353>)].

Actually I use Version 1.17.0-64bit and 1.18.0-64bit on Windows 10.

Further tests showed, that it is not possible to assign a constant value > 2^31 - 1 to a variable of type Integer.
`Integer z;`
`z := 12147483647`
Results in:
_[C:/problems.mo:50:8-50:19:writable] Warning: OpenModelica (64-bit) only supports 63-bit signed integers! Transforming: 12147483647 into a real
[C:/problems.mo:50:3-50:19:writable] Error: Type mismatch in assignment in z := 12147483647.0 of Integer := Real
[C:/problems.mo:50:3-50:19:writable] Error: Type mismatch in pattern z_

If I start with a value less than the mentioned, it can grow up to 2^63 - 1:
`Integer z;`
`z := 2147483647; // 2^31 - 1`
`z := z * 2`

But if I use `String(z)` or `intDiv(z,2)` I will get wrong (negativ) results:
`z := 2147483647;`
`z := z +10;`
`print(String(z));` gives  `-2147483639`
`print(String(intDiv(z,2)));` gives  `-1073741819`
It looks like a 32bit result of a 64bit value.

You can see this behaviour also in OMEdit, using the debugger.

**Comment by adrpo** [comment]
Yeah, I also noticed that we get the 32 bit integer message on 64 bit systems. I'll have a look.
